bpfd-operator/controllers/bpfd-agent/bpfd-agent.go

```diff
@@ -241,147 +251,141 @@ func (r *BpfProgramReconciler) reconcileBpfProgramConfig(ctx context.Context,
 
 	isNodeSelected = selector.Matches(nodeLabelSet)
 
-	programKey := internal.ProgramKey{
-		Name:        BpfProgramConfig.Spec.Name,
-		ProgType:    BpfProgramConfig.Spec.Type,
-		AttachPoint: internal.StringifyAttachType(&BpfProgramConfig.Spec.AttachPoint),
-	}
+	// Convert BpfProgramConfig.Spec object to a list of BpfProgramConfig.Spec, which maps any non-interface
+	// (like pod label or primary interface flag) into its actual interface or list of interfaces.
+	bpfProgramConfigSpecList := internal.ConvertToBpfProgramConfigSpecList(&BpfProgramConfig.Spec, r.NodeIface)
```

Comment:
I'm still not stoked on this... It really explodes the complexity of this function... I get having to expand an interface selector to a list of interfaces (with a list of bpfProgramConfigs) but now we're reconciling essentially multiple config objects in `reconcileBpfProgramConfig` which is a function designed to explicitly reconcile a single one. And from my experience we're going to run into unforseen races here